rsopt/simulation.py

```diff
@@ -164,9 +164,10 @@ def __call__(self, H, persis_info, sim_specs, libE_info):
                 f_post(self.J)
 
         if self.J['sim_status'] == message_numbers.WORKER_DONE and not halt_job_sequence:
-            # Use objective function is present
-            if self.objective_function:
-                val = self.objective_function(self.J)
+            # Use objective function if given
+            _obj_f = rsopt.util.get_objective_function(self.objective_function)
+            if _obj_f:
+                val = _obj_f(self.J)
                 output = format_evaluation(self.sim_specs, val)
                 self.log.info('val: {}, output: {}'.format(val, output))
             else:
```

Comment:
Are there any other cases besides serial python that don't need an objective? Do we need a case instead of a try, essentially?